app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
+            map.setMyLocationEnabled(true);
+        }
     }
 
     @Override
     protected void onResume() {
         super.onResume();
-        mMyLocationOverlay.enableMyLocation();
-        mMyLocationOverlay.enableCompass();
-    }
 
-    protected boolean isRouteDisplayed() {
-        return false;
+        if (mMap != null && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+            mMap.setMyLocationEnabled(true);
+        }
     }
 
-    private Location getLocation() {
-        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_DENIED ||
-                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_DENIED) {
-            // TODO PLM: warn user and ask for permissions if the user has disabled them
-            return null;
-        }
+    @Override
+    protected void onPause() {
+        super.onPause();
 
-        // Get the location manager
-        mLocationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
-        // Define the criteria how to select the locatioin provider -> use
-        // default
-        Criteria criteria = new Criteria();
-        String provider = mLocationManager.getBestProvider(criteria, false);
-        return mLocationManager.getLastKnownLocation(provider);
+        if (mMap != null && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+            mMap.setMyLocationEnabled(false);
+        }
     }
 
-    private void setLocationBounds(Location location, double[] boundHints) {
-        // Initialize the location fields
-        if (location != null) {
-            double lat = location.getLatitude();
-            double lng = location.getLongitude();
-
-            //lLat
-            boundHints[0] = lat -1;
-            //lLng
-            boundHints[1] = lng -1;
+    @Override
+    public void onInfoWindowClick(Marker marker) {
+        Intent i = new Intent(EntityMapActivity.this.getIntent());
+        TreeReference ref = markerReferences.get(marker);
+        SerializationUtil.serializeToIntent(i, EntityDetailActivity.CONTEXT_REFERENCE, ref);
 
-            //uLat
-            boundHints[2] = lat + 1;
+        setResult(RESULT_OK, i);
+        EntityMapActivity.this.finish();
+    }
 
-            //rLon
-            boundHints[3] = lng + 1;
+    private EvaluationContext getEvaluationContext() {

```

Comment:
This isn't needed: since `getEvaluationContext` is only called in `onCreate` you can make `entityEvaluationContext` a local variable and set it once like `entityEvaluationContext = session.getEvaluationContext(getInstanceInit());`